airflow/models/taskinstance.py

```diff
@@ -2555,7 +2811,7 @@ def _run_raw_task(
                 # a trigger.
                 if raise_on_defer:
                     raise
-                self.defer_task(defer=defer, session=session)
+                self.defer_task(exception=defer, session=session)
```

Comment:
I believe this whole `_run_raw_task` implementation should not be duplicated here? We should be able to provide one implementation that will serve both DB isolation/No DB isolation cases.

Reply:
doh! oversight.  _now_ let's see how many tests fail!

Reply:
done